src/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/ValueToItems.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.Tools.BootstrapTasks
+{
+	public class ValueToItems : Task
+	{
+		[Required]
+		public  string              Value                   { get; set; }
+
+		[Required]
+		public  string              Separator               { get; set; }
+
+		public  ITaskItem[]         Exclude                 { get; set; }
+
+		public  string[]            AdditionalMetadata      { get; set;}
+
+		[Output]
+		public  ITaskItem[]         Items                   { get; set; }
+
+		public override bool Execute ()
+		{
+			Log.LogMessage (MessageImportance.Low, $"Task {nameof (ValueToItems)}");
+			Log.LogMessage (MessageImportance.Low, $"  {nameof (Value)}: {Value}");
+			Log.LogMessage (MessageImportance.Low, $"  {nameof (Separator)}: '{Separator}'");
+			Log.LogMessage (MessageImportance.Low, $"  {nameof (Exclude)}:'");
+			foreach (var e in (Exclude ?? new ITaskItem[0])) {
+				Log.LogMessage (MessageImportance.Low, $"    {e}");
+			}
+			Log.LogMessage (MessageImportance.Low, $"  {nameof (AdditionalMetadata)}:'");
+			foreach (var e in (AdditionalMetadata ?? new string [0])) {
+				Log.LogMessage (MessageImportance.Low, $"    {e}");
+			}
+
+			var md          = AdditionalMetadata
+				?.Select (m => m.Split (new [] { '=' }, count: 2))
+				?.ToList ();
+
+			var items       = new List<ITaskItem> ();
+			var excludes    = Exclude?.Select (e => e.ItemSpec)?.ToList ();
+
+			var values      = Value.Split (new [] { Separator }, StringSplitOptions.RemoveEmptyEntries);
+			foreach (var v in values) {
+				if (excludes != null && excludes.Any (e => string.Equals (e, v, StringComparison.OrdinalIgnoreCase)))
+					continue;
+				var item    = new TaskItem (v);
```

Comment:
`TaskItem` has a [constructor](https://msdn.microsoft.com/en-us/library/ms126298.aspx) which accepts a `IDictionary` for the metadata, so, you don't need to manually add it every time.